src/main/scala/io/opentargets/etl/common/DrugUtils.scala

```diff
@@ -0,0 +1,86 @@
+package io.opentargets.etl.common
+
+import org.apache.spark.sql.DataFrame
+import org.apache.spark.sql.functions._
+
+object DrugUtils {
+
+  /** Obtains a lookup table of drug id and CHEBI id
+    * @param drugsDF
+    *   Data Frame containing drug information
+    * @return
+    *   Lookup table with drug id and CHEBI id
+    */
+  private def getDrugChebiIdLut(drugsDF: DataFrame): DataFrame =
+    drugsDF
+      .select(col("id"), explode(col("crossReferences")))
+      .filter(col("key") === "chEBI")
+      .withColumn("drugId", explode(col("value")))
+      .select(
+        col("id").alias("drugIdCrossChebi"),
+        concat(lit("CHEBI_"), col("drugId")).alias("drugFromSourceId")
+      )
+      .distinct()
+
+  private def completeByChebi(mapToDF: DataFrame, moleculeDF: DataFrame): DataFrame = {
+    val chebiLut = getDrugChebiIdLut(moleculeDF)
+
+    mapToDF.join(chebiLut, Seq("drugFromSourceId"), "left")
+  }
+
+  /** Obtains a lookup table of drug name and ids and it takes only oen id when more than one is
```

Comment:
This is the case we encounter when there was more than one chemblid for the same drug name. You created an issue for CHEMBL because this shouldn't happen and we said in the meantime we would take the shortest one.